File: Labwork_first/src/main/java/gui/Gui.java
package gui;

import javax.swing.*;

public class Gui {

    private final Integer monitor = Integer.valueOf("1");

    private final JFrame jFrame = new JFrame();
    private final JPanel jPanel = new JPanel();
    private final JProgressBar jProgressBar = new JProgressBar();

    private JButton firstStartButton;
    private JButton secondStartButton;

    private Thread thread1;
    private Thread thread2;

    private final Runnable runnable1 = () -> {
        synchronized (monitor) {
            while (!Thread.currentThread().isInterrupted()) {
                jProgressBar.setValue(10);
            }
        }
    };

    private final Runnable runnable2 = () -> {
        synchronized (monitor) {
            while (!Thread.currentThread().isInterrupted()) {
                jProgressBar.setValue(90);
            }
        }
    };

    public Gui() {
        performJFrameSetup();
        performJPanelSetup();
    }

    private void performJFrameSetup() {
        jFrame.setVisible(true);
        jFrame.setSize(500, 200);
        jFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        jFrame.add(jPanel);
    }

    private void performJPanelSetup() {
        jPanel.add(jProgressBar);
        jPanel.add(firstStartButton = createStartFirstThreadButton());
        jPanel.add(secondStartButton = createStartSecondThreadButton());
        jPanel.add(createStopFirstThreadButton());
        jPanel.add(createStopSecondThreadButton());
    }

    private JButton createStartFirstThreadButton() {
        JButton startButton = new JButton("Start 1");

        startButton.addActionListener(e -> {
            startButton.setEnabled(false);
            thread1 = new Thread(runnable1);
            thread1.setPriority(Thread.MIN_PRIORITY);
            thread1.start();
        });

        return startButton;
    }

    private JButton createStartSecondThreadButton() {
        JButton startButton = new JButton("Start 2");

        startButton.addActionListener(e -> {
            startButton.setEnabled(false);
            thread2 = new Thread(runnable2);
            thread2.setPriority(Thread.MAX_PRIORITY);
            thread2.start();
        });

        return startButton;
    }

    private JButton createStopFirstThreadButton() {
        JButton stopButton = new JButton("Stop 1");

        stopButton.addActionListener(e -> {
            thread1.interrupt();
            firstStartButton.setEnabled(true);
        });

        return stopButton;
    }

    private JButton createStopSecondThreadButton() {
        JButton stopButton = new JButton("Stop 2");

        stopButton.addActionListener(e -> {
            thread2.interrupt();
            secondStartButton.setEnabled(true);
        });

        return stopButton;
    }
}

File: Labwork_first/src/main/kotlin/Gui.kt
import javax.swing.*


class Gui {

    private val fJFrame: JFrame = JFrame()
    private val fJPanel: JPanel = JPanel()
    private val fJProgressBar: JProgressBar = JProgressBar()

    private val runnable1 = {
        while (!Thread.currentThread().isInterrupted) {
            fJProgressBar.value = 10
        }
    }

    private val runnable2 = {
        while (!Thread.currentThread().isInterrupted) {
            fJProgressBar.value = 90
        }
    }

    private lateinit var thread1: Thread
    private lateinit var thread2: Thread

    init {
        performJFrameSetup()
        performJPanelSetup()
    }

    private fun performJFrameSetup() {
        fJFrame.isVisible = true
        fJFrame.setSize(200, 200)
        fJFrame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        fJFrame.add(fJPanel)
    }

    private fun performJPanelSetup() {
        fJPanel.add(fJProgressBar)
        fJPanel.add(createStartButton())
        fJPanel.add(createStopButton())
        performPrioritySelectorsSetup()
    }

    private fun createStartButton(): JButton {
        val startButton = JButton("Start")

        startButton.addActionListener {
            thread1 = Thread(runnable1)
            thread1.start()

            thread2 = Thread(runnable2)
            thread2.start()
        }

        return startButton
    }

    private fun createStopButton(): JButton {
        val stopButton = JButton("Stop")

        stopButton.addActionListener {
            thread1.interrupt()
            thread2.interrupt()
        }

        return stopButton
    }

    private fun performPrioritySelectorsSetup() {
        val items = Severity.values()

        val threadFirstPriority = JComboBox(items)
        threadFirstPriority.addActionListener {
            when (threadFirstPriority.selectedItem) {
                Severity.LOW -> thread1.priority = Thread.MIN_PRIORITY
                Severity.MEDIUM -> thread1.priority = Thread.NORM_PRIORITY
                Severity.HIGH -> thread1.priority = Thread.MAX_PRIORITY
            }
        }

        val threadSecondPriority = JComboBox(items)
        threadSecondPriority.addActionListener {
            when (threadSecondPriority.selectedItem) {
                Severity.LOW -> thread2.priority = Thread.MIN_PRIORITY
                Severity.MEDIUM -> thread2.priority = Thread.NORM_PRIORITY
                Severity.HIGH -> thread2.priority = Thread.MAX_PRIORITY
            }
        }

        fJPanel.add(threadFirstPriority)
        fJPanel.add(threadSecondPriority)
    }

    private enum class Severity {
        LOW, MEDIUM, HIGH
    }
}
